core/src/main/java/org/apache/accumulo/core/file/rfile/bcfile/Compression.java

```diff
@@ -147,25 +148,27 @@ public synchronized OutputStream createCompressionStream(OutputStream downStream
     },
 
     GZ(COMPRESSION_GZ) {
-      private transient DefaultCodec codec;
+      private transient AtomicReference<DefaultCodec> codec = new AtomicReference<DefaultCodec>();
```

Comment:
I don't know why the original author did this. I didn't think so but there may be some reason? The part that gets serialized is a string, not the codec obj itself.